hooks/open-telemetry/pkg/metrics_test.go

```diff
@@ -0,0 +1,405 @@
+
+	for _, m := range metrics {
+		if m.Name == evaluationActive {
+			activeEvalMetric = m
+			break
+		}
+	}
+
+	if reflect.ValueOf(activeEvalMetric).IsZero() {
+		t.Errorf("expected %s to be present", evaluationActive)
+	}
+
+	m := activeEvalMetric.Data.(metricdata.Sum[int64])
+
+	// verify for zero count with before + finally execution
+	if m.DataPoints[0].Value != 0 {
+		t.Errorf("expected 0 value with before & finally stage executions")
+	}
+}
+
+func TestMetricHook_OptionMetadataDimensions(t *testing.T) {
+	// Test validates WithFlagMetadataDimensions option
+
+	// given
+	manualReader := metric.NewManualReader()
+	ctx := context.Background()
+
+	scopeDescription := DimensionDescription{
+		Key:  "scope",
+		Type: String,
+	}
+	scopeValue := "7c34165e-fbef-11ed-be56-0242ac120002"
+
+	stageDescription := DimensionDescription{
+		Key:  "stage",
+		Type: Int,
+	}
+	stageValue := 1
+
+	scoreDescription := DimensionDescription{
+		Key:  "score",
+		Type: Float,
+	}
+	scoreValue := 4.5
+
+	cachedDescription := DimensionDescription{
+		Key:  "cached",
+		Type: Bool,
+	}
+	cacheValue := false
+
+	evalMetadata := map[string]interface{}{
+		scopeDescription.Key:  scopeValue,
+		stageDescription.Key:  stageValue,
+		scoreDescription.Key:  scoreValue,
+		cachedDescription.Key: cacheValue,
+	}
+
+	evalDetails := openfeature.InterfaceEvaluationDetails{
+		Value: true,
+		EvaluationDetails: openfeature.EvaluationDetails{
+			FlagKey:  "flagA",
+			FlagType: openfeature.Boolean,
+			ResolutionDetail: openfeature.ResolutionDetail{
+				FlagMetadata: evalMetadata,
+			},
+		},
+	}
+	hookHints := openfeature.NewHookHints(map[string]interface{}{})
+
+	metricsHook, err := NewMetricsHook(manualReader,
+		WithFlagMetadataDimensions(scopeDescription, stageDescription, scoreDescription, cachedDescription))
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	// when
+	err = metricsHook.After(ctx, hookContext(), evalDetails, hookHints)
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	// then
+	var data metricdata.ResourceMetrics
+
+	err = manualReader.Collect(ctx, &data)
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	scopeMetrics := data.ScopeMetrics
+	if len(scopeMetrics) < 1 {
+		t.Error("expected scope metrics to be non empty with after hook")
+	}
+
+	metrics := scopeMetrics[0].Metrics
+	if len(metrics) < 1 {
+		t.Errorf("expected metric, %s to be present with after hook", evaluationSuccess)
+	}
+
+	successMetric := metrics[0]
+
+	if successMetric.Name != evaluationSuccess {
+		t.Errorf("expected %s to be present with after hook", evaluationSuccess)
+	}
+
+	instrument := successMetric.Data.(metricdata.Sum[int64])
+
+	if len(instrument.DataPoints) < 1 {
+		t.Error("expected data points, but found none")
+	}
+
+	attributes := instrument.DataPoints[0].Attributes
+
+	value, ok := attributes.Value(attribute.Key(scopeDescription.Key))
+	if !ok || value.AsString() != scopeValue {
+		t.Errorf("attribute %s is incorrectoly configured", scoreDescription.Key)
```

Comment:
I know it is a nit but there is a repeated typo here :) 
```suggestion
		t.Errorf("attribute %s is incorrectly configured", scoreDescription.Key)
```

Reply:
Really good observation 🙇